### DISCUSSION TITLE
Is it possible to override a commit message?

### DISCUSSION THREAD

**Discussion by coaxial** [semantic-release/semantic-release#2764] [Commit Convention]
I'm thinking about what `release-please` does where it is possible to override the commit message.

For instance, if a PR has been merged and the commit message on `master` is `Improve the Flubernator`, this will cause issues because it's not a semantic commit. With `release-please`, it is possible to edit the PR's body and put the semantic commit there instead so that when determining how to handle the release, it will use that description instead of the invalid commit message[^0].

Does this package provide a similar functionality? If not, what is the recommended way to fix invalid commit messages that have been merged without rewriting `master`'s history?

[^0]: https://github.com/googleapis/release-please#how-can-i-fix-release-notes

**Comment by Tlepel** [comment]
We use [commitlint](https://commitlint.js.org/#/) in a `commit-msg` git hook to verify that all commits in a PR comply with the correct formatting required for `semantic-release`. Both on local machines when developing and in the PR (using GitHub Action or GitLab Pipeline). Is that something that you could use as well?

I'm not sure about rewriting the history before you started to use `semantic-release`, I don't think git supports editing commit messages without rewriting the history (and subsequently needing to rebase all open PRs on the 'new' `master`)
